packages/base/src/processing.ts

```diff
@@ -0,0 +1,207 @@
+  if (!selected || !tracker.currentWidget) {
+    return;
+  }
+
+  const model = tracker.currentWidget.model;
+  const sources = model?.sharedModel.sources ?? {};
+  // const layers = model?.sharedModel.layers ?? {};
+
+  const geojsonString = await getLayerGeoJSON(selected, sources, model);
+  if (!geojsonString) {
+    return;
+  }
+
+  const schema = {
+    ...(formSchemaRegistry.getSchemas().get(processingType) as IDict)
+  };
+  const selectedLayerId = Object.keys(
+    model?.sharedModel.awareness.getLocalState()?.selected?.value || {}
+  )[0];
+
+  // Open ProcessingFormDialog
+  const formValues = await new Promise<IDict>(resolve => {
+    const dialog = new ProcessingFormDialog({
+      title: processingType.charAt(0).toUpperCase() + processingType.slice(1),
+      schema,
+      model,
+      sourceData: { inputLayer: selectedLayerId },
+      formContext: 'create',
+      processingType,
+      syncData: (props: IDict) => {
+        resolve(props);
+        dialog.dispose();
+      }
+    });
+    dialog.launch();
+  });
+
+  if (!formValues) {
+    return;
+  }
+
+  const processParam =
+    processingType === 'Buffer'
+      ? formValues.bufferDistance
+      : formValues.dissolveField;
+  await executeSQLProcessing(
+    model,
+    geojsonString,
+    sqlQueryFn,
+    processParam,
+    selected.name,
+    processingType
+  );
+}
+
+/**
+ * Extract GeoJSON from selected layer's source
+ */
+export async function getLayerGeoJSON(
+  layer: IJGISLayer,
+  sources: IDict,
+  model: IJupyterGISModel
+): Promise<string | null> {
+  if (!layer.parameters || !layer.parameters.source) {
+    console.error('Selected layer does not have a valid source.');
+    return null;
+  }
+
+  const source = sources[layer.parameters.source];
+  if (!source || !source.parameters) {
+    console.error(
+      `Source with ID ${layer.parameters.source} not found or missing path.`
+    );
+    return null;
+  }
+
+  return await getGeoJSONDataFromLayerSource(source, model);
+}
+
+/**
+ * Execute SQL processing on the layer
+ */
+export async function executeSQLProcessing(
+  model: IJupyterGISModel,
+  geojsonString: string,
+  sqlQueryFn: (layerName: string, param: any) => string,
+  processParam: any,
+  layerNamePrefix: string,
+  processingType: 'Buffer' | 'Dissolve'
+) {
+  const geoFile = new File(
+    [new Blob([geojsonString], { type: 'application/geo+json' })],
+    'data.geojson',
+    {
+      type: 'application/geo+json'
+    }
+  );
+
+  const Gdal = await getGdal();
+  const result = await Gdal.open(geoFile);
+
+  if (result.datasets.length === 0) {
+    return;
+  }
+
+  const dataset = result.datasets[0] as any;
+  const layerName = dataset.info.layers[0].name;
+  const sqlQuery = sqlQueryFn(layerName, processParam);
+
+  const options = [
+    '-f',
+    'GeoJSON',
+    '-dialect',
+    'SQLITE',
+    '-sql',
+    sqlQuery,
+    'output.geojson'
+  ];
+
+  const outputFilePath = await Gdal.ogr2ogr(dataset, options);
```

Comment:
If you make it a parameter of the function `gdalFunction`, you can use it here like:

```suggestion
  const outputFilePath = await Gdal[gdalFunction](dataset, options);
```

And buffer and dissolve would pass `gdalFunction="ogr2ogr"`

Reply:
This is done, thanks @martinRenou :)